src/openforms/js/components/admin/form_design/registrations/objectsapi/GenericObjectsApiVariableConfigurationEditor.js

```diff
@@ -0,0 +1,160 @@
+      }
+
+      return response.data;
+    },
+    // Load only once:
+    []
+  );
+
+  const getTargetPath = pathSegment => targetPaths.find(t => isEqual(t.targetPath, pathSegment));
+
+  const choices =
+    loading || error
+      ? LOADING_OPTION
+      : targetPaths.map(t => [JSON.stringify(t.targetPath), <TargetPathDisplay target={t} />]);
+
+  if (error)
+    return (
+      <ErrorMessage>
+        <FormattedMessage
+          description="Objects API variable registration configuration API error"
+          defaultMessage="Something went wrong when fetching the available target paths"
+        />
+      </ErrorMessage>
+    );
+  return (
+    <>
+      <FormRow>
+        <Field
+          name={`${namePrefix}.targetPath`}
+          label={
+            <FormattedMessage
+              defaultMessage="JSON Schema target"
+              description="'JSON Schema target' label"
+            />
+          }
+          disabled={isGeometry}
+        >
+          <TargetPathSelect
+            name={`${namePrefix}.targetPath`}
+            index={index}
+            choices={choices}
+            mappedVariable={mappedVariable}
+            disabled={isGeometry}
+          />
+        </Field>
+      </FormRow>
+      <div style={{marginTop: '1em'}}>
+        <a href="#" onClick={e => e.preventDefault() || toggleJsonSchemaVisible()}>
+          <FormattedMessage
+            description="Objects API variable configuration editor JSON Schema visibility toggle"
+            defaultMessage="Toggle JSON Schema"
+          />
+        </a>
+        {jsonSchemaVisible && (
+          <pre style={{marginTop: '1em'}}>
+            {loading || !mappedVariable.targetPath ? (
+              <FormattedMessage description="Not applicable" defaultMessage="N/A" />
+            ) : (
+              JSON.stringify(getTargetPath(mappedVariable.targetPath).jsonSchema, null, 2)
+            )}
+          </pre>
+        )}
+      </div>
+    </>
+  );
+};
+
+const TargetPathSelect = ({name, index, choices, mappedVariable, disabled}) => {
+  // To avoid having an incomplete variable mapping added in the `variablesMapping` array,
+  // It is added only when an actual target path is selected. This way, having the empty
+  // option selected means the variable is unmapped (hence the `arrayHelpers.remove` call below).
+  const {
+    values: {variablesMapping},
+    getFieldProps,
+    setFieldValue,
+  } = useFormikContext();
+  const props = getFieldProps(name);
+  const isNew = variablesMapping.length === index;
+
+  return (
+    <FieldArray
+      name="variablesMapping"
+      render={arrayHelpers => (
+        <Select
+          id="targetPath"
+          name={name}
+          allowBlank
+          choices={choices}
+          {...props}
+          disabled={disabled}
+          value={JSON.stringify(props.value)}
+          onChange={event => {
+            if (event.target.value === '') {
+              arrayHelpers.remove(index);
```

Comment:
When you have multiple fields filled in, and you clear one, all fields will be cleared. This happens because you remove the entire dataset of `index`.

I'll try to make a proof of concept solution...

Reply:
Its a bit big, and we can probability clean it up a bit, but this works.

So, when `event.target.value === ''`, we check if there are multiple selected values. We get all the keys of the mappedVariable, and check if there are more than 1 specific fields present (i.e., more that 1 field has a selected value). If there are multiple specific values, we set the current field to undefined, and otherwise we remove the entire arrayHelpers index
```suggestion
              if (
                Object.keys(mappedVariable).filter(variable =>
                  [
                    'postcodeTargetPath',
                    'houseLetterTargetPath',
                    'houseNumberTargetPath',
                    'houseNumberAdditionTargetPath',
                    'cityTargetPath',
                    'streetNameTargetPath',
                  ].includes(variable)
                ).length > 1
              ) {
                setFieldValue(name, undefined);
              } else {
                arrayHelpers.remove(name);
              }
```

Reply:
I think you should also move this `onChange` logic to a separate function, to keep thinks clear :)

Reply:
I think this is what you're referring to @vaszig ?

This is/was deliberate - because dealing with the backend validation errors on the variables tab is far from straight forward, the idea was that only knowingly valid items may be present in the mapping array. So if one configuration aspect being cleared means the item becomes invalid and would not pass backend validation, it is removed from the mapping entirely.

While in the modal that may seem not obvious, but when saving the modal the registration summary would accurately reflect that the form variable is not mapped to any registration aspect because the item is absent from the mapping, so it's not happening in an invisble way.

I may be lacking a bit more context here, so let me know if I miss anything.

Reply:
@sergei-maertens  if I understand correctly  we want to remove the item, as I have i in my PR, right?

Reply:
yes, I think so